### PR TITLE
Improved UNet Implementation for ISICs Data

### PR DESCRIPTION
**Pull request by mcost45**
Created by Mathew Costello, 45314838

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Sufficiently detailed commit messages
- Code valid for Normal Difficulty
- Test DSC > 0.8

Other Feedback
- Good code structure and documentation

**Comment by rexsw** [comment]
Code looks good for normal difficulty

Your pull request should include a description of what is in it and why it should be merged

Remember to delete commented out code that is no longer needed 

Comments such as 
    # An 'Upsampling Module', based off the 'improved UNet'.
    def upsampling_module(input, out_filter):
Do not add to the readability of the code and should not be included. But overall good comments

There’s no need to include a main function in layers_model as it is not meant to be run, this might mess things up if the file is imported in odd ways.

**Comment by mcost45** [comment]
> Code looks good for normal difficulty
> 
> Your pull request should include a description of what is in it and why it should be merged
> 
> Remember to delete commented out code that is no longer needed
> 
> Comments such as
> # An 'Upsampling Module', based off the 'improved UNet'.
> def upsampling_module(input, out_filter):
> Do not add to the readability of the code and should not be included. But overall good comments
> 
> There’s no need to include a main function in layers_model as it is not meant to be run, this might mess things up if the file is imported in odd ways.

I considered that comment as useful as it explains what the following function is used for. And I do not have a main function in layers model, only in the test driver. I check for main in layers model to make sure it isn't the file being run.

**Comment by rexsw** [comment]
> 
> I considered that comment as useful as it explains what the following function is used for. And I do not have a main function in layers model, only in the test driver. I check for main in layers model to make sure it isn't the file being run.

The name of the function already tells you what it's used for the comment doesn't explain anything you can't get from reading the function name

Sorry for not being clear but that check is what I was talking about, it's better to just leave out __main__ if the file isn't meant to be run.

**Comment by mcost45** [comment]
> > I considered that comment as useful as it explains what the following function is used for. And I do not have a main function in layers model, only in the test driver. I check for main in layers model to make sure it isn't the file being run.
> 
> The name of the function already tells you what it's used for the comment doesn't explain anything you can't get from reading the function name
> 
> Sorry for not being clear but that check is what I was talking about, it's better to just leave out **main** if the file isn't meant to be run.

Ahh I see, thank you!

File: recognition/ISICs-Improved-UNet-s4531483/test_driver.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
+import os
```
Comment (shakes76) [inline recognition/ISICs-Improved-UNet-s4531483/test_driver.py:1]: Code header block should be added for all scripts/source files

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Request Description. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.

**Pull request merged**